paper-input-behavior.html

```diff
@@ -335,6 +346,50 @@
     },
 
     /**
+     * Forward focus to inputElement
+     */
+    _onFocus: function() {
+      if (!this._shiftTabPressed) {
+        // iron-autogrow-textarea
+        if (this.inputElement.textarea) {
```

Comment:
It should probably live in both this behavior, and in the implementation for `paper-textarea`. Check out this reference for an example of what I mean: http://www.refactoring.com/catalog/replaceConditionalWithPolymorphism.html

In this case, the conditional behavior ("select the `this.inputElement` or `this.inputElement.textArea` as the native element to focus") should be split into a base implementation ("select `this.inputElement` as the native element to focus") and a sub-implementation that overrides the base ("select `this.inputElement.textArea` as the native element to focus").